pkg/store/etcd.go

```diff
@@ -176,48 +175,39 @@ func (s *Etcd) CancelWatch(key string) error {
 
 // AtomicPut put a value at "key" if the key has not been
 // modified in the meantime, throws an error if this is the case
-func (s *Etcd) AtomicPut(key string, oldValue []byte, newValue []byte, index uint64) (bool, error) {
-	resp, err := s.client.CompareAndSwap(normalize(key), string(newValue), 5, string(oldValue), 0)
+func (s *Etcd) AtomicPut(key string, value []byte, previous *KVPair) (bool, error) {
```

Comment:
@aluzzardi I think that you are not required to give each conditions but just one that it is going to be checked for the Compare and Swap. In theory the `lastIndex` is enough in our case so your call is OK (and unsure if mine was but at least giving only the `oldValue` should also be ok in `etcd`). Still we need more fine grained checks on the error:
- Is it an `etcd` related error?
- Is it an `error` related to the valued that cannot being set because the check failed? (In this case returning `store.ErrKeyModified` can help the client code take a decision like retrying the operation)

So I would just find a way to add the `store.ErrKeyModified` error back to the call because this information is useful imho.